code/Helper/Entity/Categoryhelper.php

```diff
@@ -118,8 +118,13 @@ public function getAllAttributes()
 
     public function getObject(Mage_Catalog_Model_Category $category)
     {
-        /** @var Mage_Catalog_Model_Resource_Product_Collection $productCollection */
-        $productCollection = $category->getProductCollection();
+        $storeId = $category->getStoreId();
+
+        /** @var Algolia_Algoliasearch_Helper_Entity_Producthelper $productHelper */
+        $productHelper = Mage::helper('algoliasearch/entity_producthelper');
+
+        $productCollection = clone $productHelper->getProductCollectionQuery($storeId);
+        $productCollection = $productCollection->addCategoryFilter($category);
         $productCollection = $productCollection->addMinimalPrice();
```

Comment:
Removed. And I also updated call of `getProductCollectionQuery` to not to fetch all data.